Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -84,64 +105,72 @@ class BackForwardListViewController: UIViewController, UITableViewDataSource, UI
-        }
     }
     
     func scrollTableViewToIndex(index: Int) {
+        guard index > 1 else {
+            return
+        }
         let moveToIndexPath = NSIndexPath(forRow: index-2, inSection: 0)
         self.tableView.reloadRowsAtIndexPaths([moveToIndexPath], withRowAnimation: .None)
         self.tableView.scrollToRowAtIndexPath(moveToIndexPath, atScrollPosition: UITableViewScrollPosition.Middle, animated: false)
     }
     
-    override func didRotateFromInterfaceOrientation(fromInterfaceOrientation: UIInterfaceOrientation) {
-        self.resizeHeight()
+    override func willTransitionToTraitCollection(newCollection: UITraitCollection, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+        super.willTransitionToTraitCollection(newCollection, withTransitionCoordinator: coordinator)
+        tableView.snp_updateConstraints { make in
+            make.bottom.equalTo(self.view).offset(bvc.shouldShowFooterForTraitCollection(newCollection) ? -UIConstants.ToolbarHeight : 0)
+        }
+    }
+    
+    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
+        tableView.snp_updateConstraints { make in
+            make.height.equalTo(min(CGFloat(BackForwardViewUX.RowHeight*listData.count), size.height/2))
+        }
+        if let superView = view.superview {
+            view.snp_updateConstraints { make in
+                make.edges.equalTo(superView)
+            }
+        }
     }
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        view.addSubview(shadow)
         view.addSubview(tableView)
         tableView.snp_makeConstraints { make in
             make.height.equalTo(0)
-            make.left.right.bottom.equalTo(self.view)
+            make.left.right.equalTo(self.view)
+            make.bottom.equalTo(self.view).offset(-bvc.footer.frame.height)
         }
-        self.view.layoutIfNeeded()
-        self.resizeHeight()
-        
+        shadow.snp_makeConstraints { make in
+            make.top.left.right.equalTo(self.view)
+            make.bottom.equalTo(tableView.snp_top)
+        }
+        view.layoutIfNeeded()
+        scrollTableViewToIndex(currentRow)
         setupDismissTap()
     }
     
     func setupDismissTap() {
         let tap = UITapGestureRecognizer(target: self, action: #selector(BackForwardListViewController.handleTap))
         tap.cancelsTouchesInView = false
         tap.delegate = self
-        parentViewController?.view.addGestureRecognizer(tap)
+        view.addGestureRecognizer(tap)
     }
     
     func handleTap() {
-        dismissWithAnimation()
+        dismissViewControllerAnimated(true, completion: nil)
     }
     
     func gestureRecognizer(gestureRecognizer: UIGestureRecognizer, shouldReceiveTouch touch: UITouch) -> Bool {
```

Comment:
Yeah, this is needed, the OverCurrentContext style doesn't dismiss when touched outside automatically.